docs/scenarios/jdk_8_to_17_migration.md

```diff
@@ -0,0 +1,395 @@
+- `pom.xml`
+
+## Step 1: Setup the environment
+
+### 1.1 Clone the Sample Java 8 Application
+
+Clone the migration example repository:
+
+```sh
+
+git clone https://github.com/konveyor-ecosystem/migrationex.git
+cd migrationex/migrationex
+```
+
+Navigate to File > Open in VSCode and locate the folder we just cloned and select the sub-folder `migrationex`.
+
+### 1.2 Install the Konveyor AI VS Code Extension
+
+Follow this [installation guide](https://github.com/konveyor/kai/blob/main/docs/scenarios/demo.md#configure-konveyor) to install the latest Konveyor AI IDE plugin.
+
+_Note: At the time of writing this guide, we used version 0.0.13._
+
+### 1.3 Configure Kai in VS Code
+
+1.3.1. When you launch the extension, you will land on the Welcome Page, as shown below. If the Welcome Page does not appear, proceed to the step 1.3.2.
+
+![walkthrough-1](../images/walkthrough-1.png)
+
+1.3.2. If "Set up Konveyor" is not available in the list, click the More button for additional options.
+![walkthrough-2](../images/walkthrough-2.png)
+
+1.3.3 If the welcome page does not appear, open the command palette by pressing Command + Shift + P. This will bring up a list of commands.
+![walkthrough-3](../images/walkthrough-3.png)
+
+From the command palette, locate and select the "Set up Konveyor" option. This will guide you through the configuration process.
+![walkthrough-4](../images/walkthrough-4.png)
+
+#### Configure Kai for your project.
+
+User has an option to override binaries and custom rules, however it comes with the default packaged binaries and custom rules.
+
+1. Let's select the custom rule we have
+   ![custom rule-1](../images/jdk_8_to_17_migration/customrule-1.png)
+   ![custom rule-2](../images/jdk_8_to_17_migration/customrule-2.png)
+   Make sure to select `Yes` to use the default rulesets
+   ![custom rule-3](../images/jdk_8_to_17_migration/customrule-3.png)
+   You will see a notification that custom rules updated
+   ![custom rule-4](../images/jdk_8_to_17_migration/customrule-4.png)
+
+2. Configuring analysis arguments is necessary to determine which rules apply to the project during analysis. Set up analysis arguments specific to your project by selecting the appropriate options and pressing "OK" to save the changes.
+
+We will analyze the application using the following migration targets to identify potential areas for improvement:
+
+- Openjdk11
+- Openjdk 17
+- Jakarta-ee
+
+![select_target.png](../images/jdk_8_to_17_migration/select_target.png)
+
+3. Next, set up the Generative AI key for your project. This step will open the `provider-settings.yaml` file. By default, it is configured to use OpenAI. To change the model, update the anchor `&active` to the desired block. Modify this file with the required arguments, such as the model and API key, to complete the setup. Sample of the provider-settings.yaml can be found [here](https://github.com/konveyor/editor-extensions/blob/main/vscode/resources/sample-provider-settings.yaml).
```

Comment:
I think this could use a screen shot and recommendation to help someone see what to setup.
We could redact any API keys prior to screenshot.

Ideal is we can run this example with a local model.  

At minimum we need to up at top of this document under pre-reqs suggest what model we ran this with so folks have an idea.

Reply:
updated with a screenshot